### PR TITLE
Fix: check variable type of port before trying to split

### PR DESCRIPTION
**Pull request by joshland**
I have been trying to use Ansible for SE Linux Port management, and it crashes if I used numbered ports.

### CODE REVIEW

**Comment by joshland** [comment]
Here is the ansible output from when i ran into the bug:

```
{"changed": false, "module_stderr": "Shared connection to localhost closed.
", "module_stdout": "Traceback (most recent call last):
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 102, in <module>
    _ansiballz_main()
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.system.seport', init_globals=None, run_name='__main__', alter_sys=True)
  File \"/usr/lib64/python3.7/runpy.py\", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File \"/usr/lib64/python3.7/runpy.py\", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File \"/usr/lib64/python3.7/runpy.py\", line 85, in _run_code
    exec(code, run_globals)
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 311, in <module>
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 301, in main
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 212, in semanage_port_add
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1149, in add
    self.__add(port, proto, serange, type)
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1098, in __add
    (k, proto_d, low, high) = self.__genkey(port, proto)
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1068, in __genkey
    ports = port.split(\"-\")
AttributeError: 'int' object has no attribute 'split'n
", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by stephensmalley** [comment]
Please submit patches to the selinux@vger.kernel.org mailing list via git send-email or equivalent, and be sure to add a Signed-off-by: line indicating that you are contributing the code under the terms of the upstream license and have the rights to do so (git will do this for you if you git commit --amend -s).

File: python/semanage/seobject.py [hunk 1]
```diff
@@ -1070,13 +1070,19 @@ def __genkey(self, port, proto):
...
+            pass
+        
         if len(ports) == 1:
             high = low = int(ports[0])
         else:
             low = int(ports[0])
             high = int(ports[1])
-
+            pass
+
```
Comment (stephensmalley) [inline python/semanage/seobject.py]: Yes, why are you adding pass statements?

Comment (joshland) [reply python/semanage/seobject.py]: It makes for proper code formatting when you cut-n-paste or move blocks of code. It compiles out, and I add such things without thinking.  I have been writing in python for so long that I forget why I do half the things I do.

I will remove it and resub.

**Comment by stephensmalley** [comment]
Ok, squash the commits, add a signed-off-by, and then submit to selinux@vger.kernel.org via git send-email.  If the last part is problematic for you due to email setup, request that a maintainer do so on your behalf after squashing and adding your signed-off-by line to the commit.

**Comment by joshland** [comment]
I hear you, 5x5.  I have a terrible headache, and I was whining internally about the process.  I will set this down, and I will pick it up in an hour when I have had a cup of coffee.  Thank you for examining my bug report, and for your instructions. 

I will complete the requisite steps to offer a bug fix to the mailing list when I return.

**Comment by stephensmalley** [comment]
Merged via 5bbe32a7e585dcd403739ea55a2fb25cbd184383

**Pull request closed without merging**